### PR TITLE
Update README, docs & Makefile

### PR DESCRIPTION
**Pull request by joewalsh**

### CODE REVIEW

**Comment by amrox** [comment]
👏

File: Makefile [hunk 1]
```diff
@@ -5,44 +5,22 @@ XCODE_VERSION = "$(shell xcodebuild -version 2>/dev/null)"
 help: ##Show this help
 	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//'
 
-submodules: ##Install or update submodules
-	# No-op, uncomment to re-enable submodules git submodule update --init --recursive
+setup: ##Install all requried dependencies to build & run the app
```
Comment (montehurd) [inline Makefile]: Typo `requried`

Comment (montehurd) [inline Makefile]: and below

File: README.md [hunk 2]
```diff
@@ -69,11 +72,17 @@ Certain development and maintenance tasks will require the installation of speci
 
 > **Homebrew and many other tools require the Xcode command line tools, which can be installed by running `xcode-select --install` on newer versions of OS X. They can also be installed via Xcode or downloaded from the [Apple Developer downloads page](https://developer.apple.com/downloads) on older versions of OS X.**
 
+### Carthage
+ 
+`brew install carthage`
+
+We use [Carthage](https://github.com/Carthage/Carthage) as our dependency manager. It is required to build the project. After installing carthage, (or running `scripts/setup`) you should be able to build & run in Xcode. `scripts/carthage_bootstrap` is run as a build step by Xcode. Your first build will take awhile as the dependencies are built. Subsequent builds will re-use the prebuilt dependencies.
```
Comment (montehurd) [inline README.md]: typo `awhile`

**Pull request merged**